inc/Modules/Settings/Admin.php

```diff
@@ -43,180 +61,228 @@ public function add_settings_page(): void {
-			// Redirect to edit the existing post.
-			wp_safe_redirect( admin_url( 'post.php?post=' . $existing_posts[0]->ID . '&action=edit' ) );
-			exit;
+		wp_enqueue_script( Assets::SETTINGS_SCRIPT_HANDLE );
+		wp_enqueue_style( Assets::SETTINGS_SCRIPT_HANDLE );
+
+		// only load media uploader in governing site settings page.
+		if ( ! Settings::is_governing_site() ) {
+			return;
+		}
+
+		wp_enqueue_media();
+	}
+
+	/**
+	 * Enqueue scripts and styles for the onboarding modal.
+	 */
+	private function enqueue_onboarding_scripts(): void {
+		// Bail if the site type is already set.
+		if ( ! empty( Settings::get_site_type() ) ) {
+			return;
 		}
 
-		// If no post exists, create a new one.
-		$new_post_id = wp_insert_post(
+		wp_localize_script(
+			Assets::ONBOARDING_SCRIPT_HANDLE,
+			'OneDesignSettings',
 			[
-				'post_type'    => Pattern::SLUG,
-				'post_title'   => esc_html__( 'Pattern Library', 'onedesign' ),
-				'post_content' => '<!-- wp:heading {"level":2} --><h2>Click on the "Patterns Selection" to push patterns to brand site.</h2><!-- /wp:heading -->',
-				'post_status'  => 'draft',
+				'nonce'     => wp_create_nonce( 'wp_rest' ),
+				'setup_url' => admin_url( sprintf( 'admin.php?page=%s', self::SCREEN_ID ) ),
+				'site_type' => Settings::get_site_type(), // @todo We can probably remove this.
 			]
 		);
 
-		if ( is_wp_error( $new_post_id ) ) {
-			wp_die( esc_html__( 'Error creating Pattern Library post.', 'onedesign' ) );
+		wp_enqueue_script( Assets::ONBOARDING_SCRIPT_HANDLE );
+		wp_enqueue_style( Assets::ONBOARDING_SCRIPT_HANDLE );
+	}
+
+	/**
+	 * Add body class if the modal is going to be shown.
+	 *
+	 * @param string     $classes        Existing body classes.
+	 * @param \WP_Screen $current_screen Current screen object.
+	 */
+	private function add_body_class_for_modal( string $classes, \WP_Screen $current_screen ): string {
+		if ( 'plugins' !== $current_screen->base ) {
+			return $classes;
+		}
+
+		// Bail if the site type is already set.
+		if ( ! empty( Settings::get_site_type() ) ) {
+			return $classes;
 		}
 
-		// Redirect to the newly created post for editing.
-		wp_safe_redirect( admin_url( 'post.php?post=' . $new_post_id . '&action=edit' ) );
-		exit;
+		// Add onedesign-site-selection-modal class to body.
+		$classes .= ' onedesign-site-selection-modal ';
+		return $classes;
+	}
+
+	/**
+	 * Add body class for missing sites.
+	 *
+	 * @param string     $classes Existing body classes.
+	 * @param \WP_Screen $current_screen Current screen object.
+	 */
+	private function add_body_class_for_missing_sites( string $classes, \WP_Screen $current_screen ): string {
+		// Bail if the shared sites are already set.
+		$shared_sites = Settings::get_shared_sites();
+		if ( ! empty( $shared_sites ) ) {
+			return $classes;
+		}
+
+		// Remove submenu pages.
+		remove_submenu_page( self::MENU_SLUG, CPT_Admin::PATTERN_REDIRECT_SCREEN );
+		remove_submenu_page( self::MENU_SLUG, CPT_Admin::TEMPLATE_REDIRECT_SCREEN );
+
```

Comment:
Is this still needed? If so, we should colocate it in `CPT_Admin::add_settings_page()` to enforce separation of concerns for better codesharing.